roles/translator/src/proxy/bridge.rs

```diff
@@ -200,38 +257,80 @@ impl Bridge {
+                                    );
+                                    nmn.create_notify()
+                                })
+                                .unwrap()
+                        })
+                        .unwrap();
+
+                    // If the current NewExtendedMiningJob and SetNewPrevHash are present and are
+                    // intended to be used for the same job (both messages job_id's are the same), send
+                    // the newly created `mining.notify` to the Downstream for mining.
+                    if let Some(msg) = sv1_notify_msg {
+                        warn!("handle_new_extended_mining_job sending mining.notify to Downstream");
+                        // TODO: handle the last_notify using the job_mapper instead
+                        let last_notify = self_.safe_lock(|s| s.last_notify.clone()).unwrap();
+                        last_notify
+                            .safe_lock(|s| {
+                                let _ = s.insert(msg.clone());
                             })
                             .unwrap();
-                    })
-                    .unwrap();
+                        sender_mining_notify.send(msg).await.unwrap();
 
-                // Commented out because of current Braiins Pool behavior.
-                // let sender_mining_notify =
-                //     self_.safe_lock(|s| s.sender_mining_notify.clone()).unwrap();
-                // let sv1_notify_msg = self_
-                //     .safe_lock(|s| {
-                //         s.next_mining_notify
-                //             .safe_lock(|nmn| nmn.create_notify())
-                //             .unwrap()
-                //     })
-                //     .unwrap();
-                // let sv1_notify_msg = sv1_notify_msg
-                //     .expect("Error creating `mining.Notify` from `NewExtendedMiningJob`");
-                // if let Some(msg) = sv1_notify_msg {
-                //     let last_notify = self_.safe_lock(|s| s.last_notify.clone()).unwrap();
-                //     last_notify
-                //         .safe_lock(|s| {
-                //             let _ = s.insert(msg.clone());
-                //         })
-                //         .unwrap();
-                //     sender_mining_notify.send(msg).await.unwrap();
-                // }
+                        // Flush stale jobs from job_mapper (aka retain all values greater than
+                        // this job_id)
+                        let last_stale_job_id = sv2_new_extended_mining_job.job_id;
+                        self_
+                            .safe_lock(|s| {
+                                s.job_mapper.retain(|&k, _| k > last_stale_job_id);
+                            })
+                            .unwrap();
+                    } else {
+                        panic!("NewExtendedMiningJob and SetNewPrevHash job ids mismatch");
```

Comment:
Is this how we want to handle errors in general? The rust book recommends percolating the Error back up the stack and letting the caller handle it instead of panic! My guess is it depends on what we want these binaries to be; production ready, or just examples that people have to build from?

Reply:
for now, we just panic to close the connection. i have an issue #262 that explains how we should aim to handle errors